notes/design_principles.md

```diff
@@ -0,0 +1,101 @@
+
+## Design Principles for Data Modeling in HyBox
+
+### Use discovery and search requirements as a pragmatic means to determine the needed extent of the modeling efforts
```

Comment:
... although, I do wonder if we'd want to use `dct:extent` with a relatively simple resource (bnode :frowning:) to enable that flexibility? (I'm happy to punt on that decision now, in any case.)